File: coding_test/greedy/until_one.cpp
#include <iostream>

using namespace std;
int main()
{
    int N, M;
    cin >> N >> M;
    int cnt = 0;
    // while (N != 1)
    // {
    //     if (N % M == 0)
    //     {
    //         N = N / M;
    //         cnt += 1;
    //     }
    //     else
    //     {
    //         N -= 1;
    //         cnt += 1;
    //     }
    // }
    while (1)
    {
        int target = (N / M) * M;
        ;
        cnt += N - target;
        N = target;
        if (N < M)
            break;
        N /= M;
        cnt += 1;
    }
    cnt += (N - 1);
    cout << cnt;
}

File: coding_test/greedy/until_one.py
n,k = map(int,input().split())
cnt = 0
# while 1:
#     while n%k ==0:
#         n = int(n/k)
#         cnt +=1
#         if(n==1):
#             break
#     if n ==1:
#         break
#     n -= 1
#     cnt +=1
# print(cnt)
while 1:
    target = int(n/k) *k
    cnt += (n-target)
    n = target
    if(n<k):
        break
    cnt +=1
    n//=k

cnt += n-1
print(cnt)